splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/MultiProbeTableScanOperation.java

```diff
@@ -231,34 +232,41 @@ public String toString() {
-                    .tableDisplayName(tableDisplayName)
-                    .activation(activation)
-                    .transaction(txn)
-                    .scan(scan)
-                    .template(currentTemplate)
-                    .tableVersion(tableVersion)
-                    .indexName(indexName)
-                    .reuseRowLocation(false)
-                    .keyColumnEncodingOrder(scanInformation.getColumnOrdering())
-                    .keyColumnSortOrder(scanInformation.getConglomerate().getAscDescInfo())
-                    .keyColumnTypes(getKeyFormatIds())
-                    .accessedKeyColumns(scanInformation.getAccessedPkColumns())
-                    .keyDecodingMap(getKeyDecodingMap())
-                    .rowDecodingMap(getRowDecodingMap())
-                    .baseColumnMap(baseColumnMap)
-                    .optionalProbeValue(probeValues[i])
-                    .buildDataSet(this);
-            dataSet = dataSet.union(ds);
-            i++;
+        try {
+            TxnView txn = getCurrentTransaction();
+            List<DataScan> scans = scanInformation.getScans(getCurrentTransaction(), null, activation, getKeyDecodingMap());
+            DataSet<LocatedRow> dataSet = dsp.getEmpty();
+            OperationContext<MultiProbeTableScanOperation> operationContext = dsp.<MultiProbeTableScanOperation>createOperationContext(this);
+            int i = 0;
+            for (DataScan scan : scans) {
+                deSiify(scan);
+                MultiProbeTableScanOperation clone = (MultiProbeTableScanOperation) operationContext.getClone().getOperation();
+                DataSet<LocatedRow> ds = dsp.<MultiProbeTableScanOperation, LocatedRow>newScanSet(this, tableName)
+                        .tableDisplayName(tableDisplayName)
+                        .activation(clone.getActivation())
+                        .transaction(txn)
+                        .scan(scan)
+                        .template(clone.currentTemplate)
+                        .tableVersion(tableVersion)
+                        .indexName(indexName)
+                        .reuseRowLocation(false)
+                        .keyColumnEncodingOrder(scanInformation.getColumnOrdering())
+                        .keyColumnSortOrder(scanInformation.getConglomerate().getAscDescInfo())
+                        .keyColumnTypes(getKeyFormatIds())
+                        .accessedKeyColumns(scanInformation.getAccessedPkColumns())
+                        .keyDecodingMap(getKeyDecodingMap())
+                        .rowDecodingMap(getRowDecodingMap())
+                        .baseColumnMap(baseColumnMap)
+                        .optionalProbeValue(probeValues[i])
+                        .buildDataSet(this);
+                dataSet = dataSet.union(ds);
+                i++;
+            }
+            return dataSet;
         }
-        return dataSet;
+        catch (Exception e) {
```

Comment:
Sorry for the delay, yes it is needed since clone can throw exceptions other than StandardException.  Clone is required so the activation is not shared when parallelized.